sensors/hall_effect_sensor/rpm/hall_effect_sensor.ino

```diff
@@ -0,0 +1,97 @@
+#include "hall_effect_sensor.h"
+
+
+HallEffectSensor::HallEffectSensor(int pin = 2) : Sensor(pin)
+{
+    init_sensor(pin);
+}
+
+void Sensor::init_sensor(const int pin)
+{
+    Serial.begin(9600);
+    attachInterrupt(digitalPinToInterrupt(pin), interruptFunc, RISING);
+    delay(1000); // Gather data to average.
```

Comment:
Dont delay here. This will slow down everything by a second

Reply:
This is a good point. This library should not be blocking for seconds at a time - try and implement it with timers like `millis()` instead.